src/map/script.cpp

```diff
@@ -19921,7 +19945,19 @@ BUILDIN_FUNC(instance_mapname)
  *------------------------------------------*/
 BUILDIN_FUNC(instance_id)
 {
-	script_pushint(st, script_instancegetid(st));
+	enum instance_mode mode = IM_NONE; // Default to the attached NPC
+
+	if (script_hasdata(st, 2)) {
+		mode = static_cast<instance_mode>(script_getnum(st, 2));
```

Comment:
Unspecified C++ behavior if the value is not in the range of the existing enum values.